pool-manager.js

```diff
@@ -75,76 +84,120 @@ module.exports = {
     return new PoolManager();
   },
   ejectOnErrorPoolManager: function (options) {
-    // endpoint states
-    var CLOSED            = 0,  // closed circuit: endpoint is good to use
-        HALF_OPEN_READY   = 1,  // endpoint is in recovery state: offer it for use once
-        HALF_OPEN_PENDING = 2,  // endpoint recovery is in process
-        OPEN              = 3;  // open circuit: endpoint is no good
-
-    if (!options || !(options.failureWindow && options.maxFailures && options.resetTimeout)) {
-      throw new Error('Must supply all arguments to ejectOnErrorPoolManager');
+    if (!options) {
+      throw new Error('Must supply arguments to ejectOnErrorPoolManager');
     }
 
     var failureWindow = options.failureWindow;
     var maxFailures = options.maxFailures;
     var resetTimeout = options.resetTimeout;
+    var failureRate = options.failureRate;
+    var failureRateWindow = options.failureRateWindow;
+
+    var poolConfig;
+    if (failureWindow && maxFailures && resetTimeout) {
+      poolConfig = getRollingWindowConfiguration();
```

Comment:
Not critical, but I think it's better to explicitly pass both `failureWindow` and `maxFailures` to `getRollingWindowConfiguration()` function. Otherwise, when reading the latter. it's a bit hard to understand that those variables are coming from the closure...